lib/dco.js

```diff
@@ -1,8 +1,28 @@
 module.exports = ({commit, parents}) => {
   const isMerge = parents && parents.length > 1;
-  return isMerge || commit.message.includes(signOff(commit));
+  const signed = commit.message.includes(signOff(commit));
+
+  if (isMerge || signed) {
+    return true;
+  } else if (!signed) {
+    return getError(commit);
+  }
 };
 
 function signOff(commit) {
   return `Signed-off-by: ${commit.author.name} <${commit.author.email}>`;
 }
+
+function getError(commit) {
+  let err = '';
+  if (!commit.message.includes('Signed-off-by:')) {
+    err = 'Your sign off is missing. ';
+  } else if (!commit.message.includes(commit.author.name) && !commit.message.includes(commit.author.email)) {
+    err = 'Neither author name nor email match. ';
+  } else if (!commit.message.includes(commit.author.name)) {
+    err = 'Your author name does not match. ';
+  } else if (!commit.message.includes(commit.author.email)) {
+    err = 'Your email does not match. ';
+  }
```

Comment:
So the only problem with this approach is the literal message length. So the message (as far as I know) only appears in 2 places on a pull request, at the bottom under checks, where it gets cut of past that one line, and clicking details redirects [here](https://github.com/probot/dco/blob/master/index.js#L11). The other place it shows up is if you hover over the X or ✅ it displays the message; _however_ if you have multiple checks (like here there's 2, the dco and travis CI) then it displays 1/2 checks OK.

So I'm 100% for this solution, in fact mine has a minor bug? ie my expected name and email were `hiimbex` and `hiimbex@github.com` and I tested with `othername` and `hiimbex@github.com` and the matching I was doing would count that as correct when it shouldn't have, but the ability to not view the error seems pretty critical.